scrapi/base/transformer.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import unicode_literals
+
+import abc
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class BaseTransformer(object):
+
+    __metaclass__ = abc.ABCMeta
+
+    def transform(self, doc):
+        return self._transform(self.schema, doc)
+
+    def _transform(self, schema, doc):
+        transformed = {}
+        for key, value in schema.items():
+            if isinstance(value, dict):
+                transformed[key] = self._transform(value, doc)
+            elif isinstance(value, list) or isinstance(value, tuple):
+                transformed[key] = self._transform_iter(value, doc)
+            elif isinstance(value, basestring):
+                transformed[key] = self._transform_string(value, doc)
+        return transformed
+
+    def _transform_iter(self, l, doc):
```

Comment:
Can this be `_transform_iterable`? `_transform_iter` sounds like it should return a generator.